Add nodejs toolchain

betaboon opened the pull request.

betaboon commented:
I'm still working on this.

currently i'm stuck at it running into `mismatching values: support_nix`.
i bet there is just a minor thing missing somewhere. maybe someone has a clue.
i would be greatful for any hints ;)

betaboon commented:
thanks to @chancila i figured out that only a setting in `.bazelrc` was missing:
```build:nix --extra_execution_platforms=@io_tweag_rules_nixpkgs//nixpkgs/platforms:host```

bhavitsharma commented on toolchains/nodejs/tests/BUILD.bazel:
```diff
@@ -0,0 +1,8 @@
+
+package(default_testonly = 1)
+
+nodejs_binary(
```
not sure but how is this testing any nix toolchain? Shouldn't we add some nixpkgs dependency here?

betaboon replied:
as far as i understand the test-infrastructure here, the tests are executed in the context of the `WORKSPACE` of the toolchain, which includes the `nixpkgs_go_configure`

fricklerhandwerk replied:
That is correct.

avdv commented:
Hi @betaboon , thank you for your contribution!

Are you still working on it or do you need some assistance?

betaboon commented:
@avdv I'm still working on this. planning to get the documentation ready in the next couple days.

i could need some assistance on the issue of the required `--extra_execution_platforms`, if that is something that we would want.

if there are any other pointers to get this in a ready state I'd gladly integrate them

avdv commented:
Hi @betaboon , sorry for the long delay.

I had a look at this again. Also, I rebased onto master and added a missing dependency. 

An example for the toolchain would be nice to have.

Thank you!

betaboon commented:
@avdv sorry for the long silence.

i just rebased against master, included your changes, updated the nodejs and nixpkgs rules, added an example.

i think this is now fine and ready :)

avdv commented:
Oh, I missed this update over the course of my holidays... I'm taking a look now.

avdv commented on core/.bazelrc:
```diff
@@ -1,4 +1,5 @@
 build --host_platform=@rules_nixpkgs_core//platforms:host
+build --extra_execution_platforms=@io_tweag_rules_nixpkgs//nixpkgs/platforms:host
```
Also, after reverting the commit, CI was all green: https://github.com/tweag/rules_nixpkgs/actions/runs/3095774582

betaboon replied:
iirc back then it failed to pick up the toolchain. i had some longer discussion in slack with someone that lead me to find this was required. maybe things changed since then ?
